### PR TITLE
Avoid realizing a syntax node from a tree that doesn't own the context.

### PR DESCRIPTION
**Pull request by AlekseyTs**

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs [hunk 1]
```diff
@@ -1260,7 +1260,7 @@ private SynthesizedRecordConstructor TryGetSynthesizedRecordConstructor(RecordDe
             NamedTypeSymbol recordType = GetDeclaredType(node);
             var symbol = recordType.GetMembersUnordered().OfType<SynthesizedRecordConstructor>().SingleOrDefault();
 
-            if (symbol?.GetSyntax() != node)
+            if (symbol?.SyntaxRef.SyntaxTree != node.SyntaxTree || symbol.GetSyntax() != node)
```
Comment (Youssef1313) [inline src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs:1263]: Can any behavioral change be demonstrated by a test?

Comment (AlekseyTs) [reply src/Compilers/CSharp/Portable/Compilation/SyntaxTreeSemanticModel.cs:1263]: > Can any behavioral change be demonstrated by a test?

Theoretically, yes. But I don't think it is worth the effort and we usually don't add tests like that.

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a very small change.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign-off for a very small change.

**Review by cston — APPROVED** [review]

**Pull request merged**